backend/open_webui/apps/rag/utils.py

```diff
@@ -262,19 +289,25 @@ def get_rag_context(
             continue
 
         try:
+            context = None
             if file["type"] == "text":
                 context = file["content"]
             else:
                 if hybrid_search:
-                    context = query_collection_with_hybrid_search(
-                        collection_names=collection_names,
-                        query=query,
-                        embedding_function=embedding_function,
-                        k=k,
-                        reranking_function=reranking_function,
-                        r=r,
-                    )
-                else:
+                    try:
+                        context = query_collection_with_hybrid_search(
+                            collection_names=collection_names,
+                            query=query,
+                            embedding_function=embedding_function,
+                            k=k,
+                            reranking_function=reranking_function,
+                            r=r,
+                        )
+                    except Exception as e:
+                        log.debug("Error when using hybrid search, using"
+                                    " non hybrid search as fallback.")
```

Comment:
Idk if log debug is better when sending exception
```suggestion
                        log.exception(e)
```

Reply:
Thanks for taking a look.

I decided to use debug because what's happening is
1. `query_doc_with_hybrid_search` crashes
2. we log it with `log.exception` at line 184
3. Then, only if all the collection failed
4. We raise an exception
5. That is caught at line 307
6. And shown with log.debug.

The purpose of this 2nd exception is to tell `get_rag_context` to fallback to using non hybrid search. It's not *really* and exception, especially compared to the exceptions caught at line 184.

Hence my reasoning was that the fallback is caused by an exception but we warn that we're using the fallback with a debug.

In all cases I'm not too hung up on this and can change it if anyone has a strong feeling about this.